### PR TITLE
69 signup unit tests

### LINKED ISSUE #69: Write unit tests for signUp
## Description

Adding more unit tests for signup. 

## Requirements

- [ ] Add more unit tests for signup.

### PR DESCRIPTION
**Pull request by shonancanuck**
# Description

Write unit tests for signup

# Closes issue(s)

Resolves #69

# How to test
run tests using ```npm test```

# Changes include

1. Add test and corresponding error for empty username
2. Add Add test and corresponding error for invalid email
3. Add Add test and corresponding error for empty or invalid email
4. Add test and corresponding error for UID
5. Add test for email duplicate
6. Add error when createUser returns undefined

# Checklist

- [x] I have tested this code
- [ ] I have updated the README

### CODE REVIEW

File: __tests__/signup.test.ts [hunk 1]
```diff
@@ -15,6 +15,58 @@ interface User {
 }
 
 describe('createUser', () => {
```
Comment (Anissa3005) [inline __tests__/signup.test.ts:17]: I would make the describe title more descriptive. Something like "testing error path createUser...". So when running the tests we see what is tested in the describe block.

File: src/model/signup.model.ts [hunk 2]
```diff
@@ -13,6 +10,20 @@ export default {
     email: string,
     uid: string,
   ): Promise<User> {
+    if (username.length < 1) {
```
Comment (Anissa3005) [inline src/model/signup.model.ts:13]: These are all validations and would be done in the service file. 
@oddpharmacy is currently researching what kind of validations we need to add in issue #74 since something are already validated in the schema like; is user unique, the length,... if the username is not unique prisma will throw a PrismaError and will end up in the catch block.

So for error handling I would  add a try and catch block. You can check what best practice is for prisma models. But this logic would need to be implemented in the service file since these are validation checks.

**Review by Anissa3005 — CHANGES_REQUESTED** [review]
The test them selves look great, but there are some changes needed: 

1. The title of the PR should be more descriptive, and no need for the issue number in the title since you add it in the PR comment.
2. Your branch is not up to date with main, make sure you checkout main, pull, and rebase your branch to main.
3. If I check out to your branch and run the tests one test is failing so I think it has to do with number 2. 
4. The validation logic should be removed from model. In model we would be testing prismaErrors use this as your starting point https://www.prisma.io/docs/concepts/components/prisma-client/handling-exceptions-and-errors

Maybe team up with @FefesG5 for the error messaging part. If you would add error messaging in this branch, make sure that you add it to the PR comment and the title as well. Something like "add error handling and unit test for model createUser"
